eth/protocols/eth/protocol.go

```diff
@@ -555,6 +566,24 @@ func (bb *BlockBody) DecodeRLP(s *rlp.Stream) error {
 	if err = s.ListEnd(); err != nil {
 		return err
 	}
+	// decode Withdrawals
+	if _, err = s.List(); err != nil {
+		return err
+	}
```

Comment:
Please add
```
bb.Withdrawals = []*Withdrawal{}
```
in order to handle post-Shanghai bodies with empty withdrawals (see PR #6291).